### PR TITLE
complete intellij quick start guide with some extra steps

### PR DESCRIPTION
**Pull request by sergitopo**
This PR updates one doc of the project, concretely the intellij quick start guide. Here are included now two steps that I had to take in order to run the project in a new Intellij project.
  

# Checklist

- [x ] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x ] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

### CODE REVIEW

File: doc/en/developer/source/quickstart/intellij.rst [hunk 1]
```diff
@@ -86,4 +96,7 @@ Run GeoServer from Intellij on Windows
 --------------------------------------
 
 #. Add bash to your Windows environment path and restart Intellij.  
-#. If there are errors such as "cannot find symbol AbstractUserGroupServiceTest", rebuild the security-tests project in the security module.  Right click on the security-tests project and click Rebuild.  
+#. If there are errors such as "cannot find symbol AbstractUserGroupServiceTest", rebuild the security-tests project in the security module.  Right click on the security-tests project and click Rebuild.
```
Comment (aaime) [inline doc/en/developer/source/quickstart/intellij.rst:99]: These two are not really Windows specific.

File: doc/en/developer/source/quickstart/intellij.rst [hunk 2]
```diff
@@ -35,6 +35,16 @@ Import modules into Intellij
    .. image:: img/intellij_import_finish.png
       :width: 600
 
+Before Running Geoserver
+---------------------------
+
+GeoServer relies on other libraries that are manatined in parallel under the same project umbrella. These are Geotools and GeoWebCache. So for easier installation and setup of the GeoServer development environment you need to apply the following:
```
Comment (aaime) [inline doc/en/developer/source/quickstart/intellij.rst:41]: Not strictly mandatory, although it helps to have freshly built jars. locally.

```suggestion
GeoServer relies on other libraries that are manatined in parallel under the same project umbrella. These are Geotools and GeoWebCache. So for easier installation and setup of the GeoServer development environment you may want to apply the following:
```

File: doc/en/developer/source/quickstart/intellij.rst (hunk 1 above)
Comment (aaime) [reply doc/en/developer/source/quickstart/intellij.rst:99]: Here I mean, they are good comments, but don't belong in the Windows specific section, should be above, in the general introduction.

Comment (sergitopo) [reply doc/en/developer/source/quickstart/intellij.rst]: Moved in the latest commit. So this should appear as outdated

File: doc/en/developer/source/quickstart/intellij.rst [hunk 3]
```diff
@@ -35,6 +35,16 @@ Import modules into Intellij
    .. image:: img/intellij_import_finish.png
       :width: 600
 
+Before Running Geoserver
+---------------------------
+
+GeoServer relies on other libraries that are manatined in parallel under the same project umbrella. These are Geotools and GeoWebCache. So for easier installation and setup of the GeoServer development environment you may want to apply the following:
```
Comment (aaime) [inline doc/en/developer/source/quickstart/intellij.rst:41]:
```suggestion
GeoServer relies on other libraries that are maintained in parallel under the same project umbrella. These are Geotools and GeoWebCache. So for easier installation and setup of the GeoServer development environment, you may want to apply the following:
```

File: doc/en/developer/source/quickstart/intellij.rst [hunk 4]
```diff
@@ -35,6 +35,16 @@ Import modules into Intellij
...
+
+#. Download the code of both and execute mvn clean install to them.
```
Comment (aaime) [inline doc/en/developer/source/quickstart/intellij.rst:43]:
```suggestion
#. Download the code of both and execute ``mvn clean install`` into them.
```

File: doc/en/developer/source/quickstart/intellij.rst [hunk 5]
```diff
@@ -60,6 +70,9 @@ Run GeoServer from Intellij
           :width: 400
 
     #. If you want to use ``errorprone``, notably to perform the QA checks, install the ``Error Prone Compiler`` plugin, restart the IDE and set ``Javac with error-prone`` as a default compiler for the project. Please note that this will slower the build.
+#. If there are errors such as "cannot find symbol AbstractUserGroupServiceTest", rebuild the security-tests project in the security module.  Right click on the security-tests project and click Rebuild.
```
Comment (aaime) [inline doc/en/developer/source/quickstart/intellij.rst:73]:
```suggestion
#. If there are errors such as "cannot find symbol AbstractUserGroupServiceTest", rebuild the ``security-tests`` project in the security module.  Right-click on the ``security-tests`` project and click Rebuild.
```

**Review by aaime — CHANGES_REQUESTED** [review]
Had another look and found some typos.

File: doc/en/developer/source/quickstart/intellij.rst [hunk 6]
```diff
@@ -60,6 +70,9 @@ Run GeoServer from Intellij
           :width: 400
 
     #. If you want to use ``errorprone``, notably to perform the QA checks, install the ``Error Prone Compiler`` plugin, restart the IDE and set ``Javac with error-prone`` as a default compiler for the project. Please note that this will slower the build.
+#. If there are errors such as "cannot find symbol AbstractUserGroupServiceTest", rebuild the ``security-tests`` project in the security module.  Right-click on the ``security-tests`` project and click Rebuild.
+#. In the last versions of Intellij Annotations processors are enabled. If there are errors because of this uncheck this options from compiler settings.
+    .. image::img/intellij_disable_annotation_processors.jpeg
```
Comment (jodygarnett) [inline doc/en/developer/source/quickstart/intellij.rst:75]: This image directive is indented by one character (so it will appear as a block quote)

**Comment by aaime** [comment]
Merged, thanks!

**Pull request merged**